### PR TITLE
feat: add language packages with minor pinning by default

### PR DESCRIPTION
**Pull request by ruben-arts**
Old:
```
> pixi add python
✔ Added python >=3.13.0,<4
```
New:
```
> pixi add python
✔ Added python >=3.13.0,<3.14
```

### CODE REVIEW

File: src/cli/add.rs [hunk 1]
```diff
@@ -380,18 +380,26 @@ fn update_conda_specs_from_lock_file(
...
 
-    let pinning_strategy = project.config().pinning_strategy.unwrap_or_default();
+    let mut pinning_strategy = project.config().pinning_strategy;
     let channel_config = project.channel_config();
     for (name, (spec_type, spec)) in conda_specs_to_add_constraints_for {
-        let version_constraint = pinning_strategy.determine_version_constraint(
-            conda_records.iter().filter_map(|record| {
+        // Edge case: python is a special case where we want to pin the minor version by default.
+        // This is done to avoid early user confusion when the minor version changes and environments magically start breaking.
+        // This move a `>=3.13, <4` to a `>=3.13, <3.14` constraint.
```
Comment (Hofer-Julian) [inline src/cli/add.rs:388]: Might be worth adding a docs entry for this

**Review by Hofer-Julian — COMMENTED** [review]
Special casing `python` sounds good to me.

My feeling is that we want to do that for all language toolchains.
So the same thing for `gcc`, `julia`, `rust`,...

File: src/cli/add.rs [hunk 2]
```diff
@@ -380,18 +380,26 @@ fn update_conda_specs_from_lock_file(
...
+        if name.as_normalized() == "python" && pinning_strategy.is_none() {
```
Comment (tdejager) [inline src/cli/add.rs:389]: Could we maybe make an array of these so that it's easier to add if for more if needed.

Comment (ruben-arts) [reply src/cli/add.rs]: Done

File: src/cli/add.rs (hunk 1 above)
Comment (ruben-arts) [reply src/cli/add.rs]: Done

**Pull request merged**